lib/analytics/ga4/relevancy_calculator.rb

```diff
@@ -0,0 +1,57 @@
+require "json"
+module Analytics
+  module Ga4
+    class RelevancyCalculator
+      def initialize(consolidated_data)
+        @consolidated_data = consolidated_data
+      end
+
+      def relevance
+        consolidated_data.map.with_index(1) { |(base_path, page_views), index|
+          [
+            elastic_search_index(base_path).to_json,
+            elastic_search_rank(base_path, index, page_views).to_json,
+          ]
+        }.flatten
+      end
+
+    private
+
+      attr_reader :consolidated_data
+
+      def elastic_search_index(base_path)
+        {
+          index: {
+            _type: "page-traffic",
+            _id: base_path,
+          },
+        }
+      end
+
+      def elastic_search_rank(base_path, index, page_views)
+        {
+          path_components: path_components(base_path),
+          rank_14: index,
+          vc_14: page_views,
+          vf_14: page_views / total_page_views.to_f,
+        }
+      end
+
+      def path_components(path)
+        result = []
+        if path.start_with?("/")
+          components = path.sub("/", "").split("/")
+          (1..components.length).each do |i|
+            result.append("/#{components.first(i).join('/')}")
+          end
+        end
+        result
+      end
+
+      def total_page_views
+        all_page_views = consolidated_data.values
+        all_page_views.reduce(:+)
```

Comment:
```suggestion
        @all_page_views ||= consolidated_data.values.reduce(:+)
```

save us having to recalculate this same number each iteration